parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/VectorizedReadBenchmarks.java

```diff
@@ -0,0 +1,266 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.benchmarks;
+
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.hadoop.ParquetReader;
+import org.apache.parquet.hadoop.example.GroupReadSupport;
+import org.apache.parquet.io.ColumnVector;
+import org.apache.parquet.io.vector.ObjectColumnVector;
+import org.apache.parquet.io.vector.RowBatch;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.IOException;
+
+import static org.apache.parquet.benchmarks.BenchmarkFiles.defaultConfiguration;
+import static org.apache.parquet.benchmarks.BenchmarkFiles.file_10M_GZIP;
+import static org.apache.parquet.benchmarks.BenchmarkFiles.flbaReadConfiguration;
+import static org.apache.parquet.benchmarks.BenchmarkFiles.readAllPrimitivesConfiguration;
+import static org.apache.parquet.benchmarks.BenchmarkFiles.readFourPrimitivesConfiguration;
+import static org.apache.parquet.benchmarks.BenchmarkFiles.readOnePrimitiveConfiguration;
+import static org.apache.parquet.hadoop.ParquetReader.builder;
+import static org.openjdk.jmh.annotations.Level.Invocation;
+import static org.openjdk.jmh.annotations.Scope.Benchmark;
+
+public class VectorizedReadBenchmarks {
+
+  @State(Benchmark)
+  public static class ReadAllObjects {
+    private GroupReadSupport groupReadSupport;
+    private ParquetReader<Group> defaultReader;
+
+    @Setup(Invocation)
+    public void setup() throws IOException {
+      groupReadSupport = new GroupReadSupport();
+      defaultReader = builder(groupReadSupport, file_10M_GZIP).withConf(defaultConfiguration).build();
+    }
+
+    @TearDown(Invocation)
+    public void teardown() throws IOException {
+      defaultReader.close();
+    }
+
+    @Benchmark
+    public void readAllObjects(Blackhole blackhole) throws IOException
+    {
+      Group group;
+      while ((group = defaultReader.read()) != null) {
+        blackhole.consume(group.getBinary("binary_field", 0));
+        blackhole.consume(group.getInteger("int32_field", 0));
+        blackhole.consume(group.getLong("int64_field", 0));
+        blackhole.consume(group.getBoolean("boolean_field", 0));
+        blackhole.consume(group.getFloat("float_field", 0));
+        blackhole.consume(group.getDouble("double_field", 0));
+        blackhole.consume(group.getBinary("flba_field", 0));
+        blackhole.consume(group.getInt96("int96_field", 0));
```

Comment:
All the `getXxx(fieldName, ordinal)` methods require a hash map lookup, which can be unfair for the original `ColumnReader` API in this benchmark.